daemon/daemon.go

```diff
@@ -808,7 +808,7 @@ func NewDaemon(config *Config, registryService *registry.Service, containerdRemo
 	sysInfo := sysinfo.New(false)
 	// Check if Devices cgroup is mounted, it is hard requirement for container security,
 	// on Linux/FreeBSD.
-	if runtime.GOOS != "windows" && !sysInfo.CgroupDevicesEnabled {
+	if runtime.GOOS != "windows" && runtime.GOOS != "solaris" && !sysInfo.CgroupDevicesEnabled {
```

Comment:
Should this test be rewritten as `runtime.GOOS == "linux" || runtime.GOOS == "freebsd"` - seems better to test on where it is required rather than listing everywhere it is not required.